rakelib/utils.rake

```diff
@@ -29,4 +46,33 @@ class String
     return false if self.empty? || self =~ (/^(false|f|no|n|0)$/i)
     raise ArgumentError.new "invalid value: #{self}"
   end
+
+  # colorization
+  def black;          colors_available? ? "\e[30m#{self}\e[0m" : self end
```

Comment:
I wonder if it's such a good idea to "pollute" the `String` type with so many functions with short and common name (risk of clash?)

What about:
```
COLORS = {
  :red => 1
  :green => 2
…
}
STYLES = {
  :bold => 1
  …
}

def colorize(fg, bg = nil, *styles)
  return self unless colors_available?
  result = self

  code = COLORS[fg]
  result = "\e[3#{code}m#{result}\e[0m" unless code.nil?
  code = COLORS[bg]
  result = "\e[4#{code}m#{result}\e[0m" unless code.nil?

  styles.each do |s|
    code = STYLES[s]
    result = "\e[#{code}m#{self}\e[2#{code}m" unless code.nil?
  end
end
```

Usage: `"Hello".colorize(:red, :blue, :bold, :italic)`